core/tests/snapshots.rs

```
@@ -760,3 +762,147 @@ fn test_snapshots_with_background_services() {
+        assert_eq!(bank.process_transaction(&tx), Ok(()));
+
+        let key = solana_pubkey::new_rand();
+        let tx = system_transaction::transfer(mint_keypair, &key, 0, bank.last_blockhash());
+        assert_eq!(bank.process_transaction(&tx), Ok(()));
+
+        bank.fill_bank_with_ticks_for_tests();
+
+        // Inject a fastboot snapshot at a specific slot
+        if slot % FASTBOOT_SNAPSHOT_INTERVAL_SLOTS == 0 {
+            bank_forks
+                .write()
+                .unwrap()
+                .set_root(slot, Some(&snapshot_controller), None);
+
+            bank.force_flush_accounts_cache();
+            let snapshot_package = SnapshotPackage::new(
+                SnapshotKind::Fastboot,
+                &bank,
+                bank.get_snapshot_storages(None),
+                bank.status_cache.read().unwrap().root_slot_deltas(),
+            );
+            pending_snapshot_packages
+                .lock()
+                .unwrap()
+                .push(snapshot_package);
+
+            // Wait while the fastboot snapshot is processed
+            while !pending_snapshot_packages.lock().unwrap().is_empty() {
+                std::thread::sleep(Duration::from_millis(100));
+            }
+        }
+    }
+
+    let bank_snapshots = snapshot_utils::get_bank_snapshots(
+        &snapshot_test_config.snapshot_config.bank_snapshots_dir,
+    );
+
+    if exit_backpressure.is_none() {
+        // Without backpressure, the fastboot snapshot should be present
+        assert!(!bank_snapshots.is_empty());
+    } else {
+        // With backpressure, the fastboot snapshot will not be written until teardown
+        // is triggered by writing true to exit below
+        assert!(bank_snapshots.is_empty());
+    }
+
+    // Stop the background services, ignore any errors
+    exit.store(true, Ordering::Relaxed);
+    let packager_exit_result = snapshot_packager_service.join();
+    assert!(packager_exit_result.is_ok());
```

The comment says errors are ignored, but the code is asserting no errors.